nativelib/src/main/scala/scala/scalanative/posix/sys/socket.scala

```diff
@@ -0,0 +1,162 @@
+package scala.scalanative.posix.sys
+
+import scala.scalanative.native.Nat._
+import scala.scalanative.native._
+import scala.scalanative.posix.sys.types.ssize_t
+import scala.scalanative.posix.sys.uio.iovec
+import scala.scalanative.runtime.time.timespec
+
+// http://man7.org/linux/man-pages/man7/socket.7.html
+
+/**
+ * Created by remi on 01/03/17.
+ */
+@extern
+object socket {
+
+  def socket(domain: CInt, _type: CInt, protocol: CInt): CInt = extern
+  def socketpair(domain: CInt,
+                 _type: CInt,
+                 protocol: CInt,
+                 sv: CArray[CInt, _2]): CInt = extern
+  def bind(sockfd: CInt, addr: Ptr[sockaddr], addrlen: socklen_t): CInt =
+    extern
+  def getsockname(sockfd: CInt,
+                  addr: Ptr[sockaddr],
+                  addrlen: Ptr[socklen_t]): CInt = extern
+  def connect(sockfd: CInt, addr: Ptr[sockaddr], addrlen: socklen_t): CInt =
+    extern
+  def getpeername(sockfd: CInt,
+                  addr: Ptr[sockaddr],
+                  addrlen: Ptr[socklen_t]): CInt = extern
+  def send(sockfd: CInt, buf: Ptr[Byte], len: CSize, flags: CInt): ssize_t =
+    extern
+  def sendto(sockfd: CInt,
+             buf: Ptr[Byte],
+             len: CSize,
+             flags: CInt,
```

Comment:
@Korf74 It's up to you. It can be in a separate PR too.

I think the `@name` macro is to make it available in C. Since it's already available via the header file I think it's redundant and can be removed.